codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/samples/KDocSamplesGenerator.kt

```diff
@@ -0,0 +1,143 @@
+     */
+    override fun preprocessModel(model: Model, settings: KotlinSettings): Model {
+        val transformer = ModelTransformer.create()
+        return transformer.mapTraits(model) { shape, trait ->
+            when {
+                shape is OperationShape && shape.hasTrait<ExamplesTrait>() && trait is DocumentationTrait -> {
+                    val examplesTrait = shape.expectTrait<ExamplesTrait>()
+                    val filtered = examplesTrait.examples.filterNot { it.error.isPresent }
+                    val kdocSampleIdentifiers = filtered.indices.joinToString(separator = "\n") { idx ->
+                        val identifier = sampleIdentifier(settings, shape, idx)
+                        "@sample $identifier"
+                    }
+
+                    val updatedDocs = trait.value + "\n\n" + kdocSampleIdentifiers
+                    DocumentationTrait(updatedDocs, trait.sourceLocation)
+                }
+                else -> trait
+            }
+        }
+    }
+
+    private fun sampleIdentifier(settings: KotlinSettings, op: OperationShape, index: Int): String =
+        listOf(
+            samplePackage(settings),
+            sampleClassName(op),
+            sampleFunctionName(index),
+        ).joinToString(separator = ".")
+
+    private fun sampleFunctionName(index: Int): String = "sample${index + 1}"
+    private fun sampleClassName(op: OperationShape): String = op.id.name
+
+    private fun samplePackage(settings: KotlinSettings): String = settings.pkg.subpackage("samples")
+    override fun writeAdditionalFiles(ctx: CodegenContext, delegator: KotlinDelegator) {
+        val topDownIndex = TopDownIndex.of(ctx.model)
+        val operations = topDownIndex.getContainedOperations(ctx.settings.service)
+        operations.filter { it.hasTrait<ExamplesTrait>() }
+            .forEach { op ->
+                val examples = op.expectTrait<ExamplesTrait>()
+
+                val writer = KotlinWriter(samplePackage(ctx.settings))
+                writer.withBlock("class #L {", "}", sampleClassName(op)) {
+                    examples.examples
+                        // unclear what benefit error examples provide, omit for now
+                        .filterNot { it.error.isPresent }
+                        .forEachIndexed { idx, example ->
+                            write("")
+                            write("@Sample")
+                            withBlock("fun #L() {", "}", sampleFunctionName(idx)) {
+                                example.documentation
+                                    .getOrDefault(example.title)
+                                    .breakLongLines()
+                                    .forEach { line ->
+                                        write("// #L", line)
+                                    }
+
+                                renderNormalExample(ctx, writer, op, example)
+                            }
+                            write("")
```

Comment:
**Question**: Why blank lines before _and_ after the function?

Reply:
```
class Foo {
    @Sample
     fun foo() { ... }
     @Sample
     fun bar() { ... }
}
```

vs

```
class Foo {
    
     @Sample
     fun foo() { ... }
     
     @Sample
     fun bar() { ... }

}
```

Reply:
Could just be before, seems really minor since nobody is going to actually look at these (only the function body is rendered into API ref documentation)